src/systemctl/systemctl.c

```diff
@@ -5272,6 +5272,17 @@ static int cat(int argc, char *argv[], void *userdata) {
                 else
                         puts("");
 
+                if (need_daemon_reload(bus, *name))
+                        printf("%s# %sWarning:%s %s changed on disk, showing current disk contents.\n"
```

Comment:
Ha, I thought that this would be a more complicated patch. It's nice when existing functionality gets reused like that.

> It prints the warning without any special formatting to stderr, which doesn’t break systemctl cat foo.service > foo2.service, but looks ugly in the terminal.

Hm, maybe we could print the warning to stdout if we were originally on the terminal (before the pager was started), and to stderr otherwise? We do a similar check in various other cases.

Why make the message blue? Shouldn't it be red as other warnings?

I'm not sure about the message:
```
# Warning: abrt-ccpp.service changed on disk, showing current disk contents.
# Run 'systemctl daemon-reload' to reload units.
```
The first line basically says that you're showing the latest version ;)
There are two cases to consider:
- an *existing* file was modified, either the main unit file, or one of the drop-ins.
- a drop in was added or removed

In the first case, `cat` is showing good content, but the manager has an outdated view.
In the second case, `cat` is showing bad content, *and* the manager has an outdated view.
I don't think we can distinguish those cases with existing functionality, so the message should be updated to reflect both. Maybe something like "...service was changed on disk, systemd has an outdated version loaded. If drop-ins were added or removed, they are not properly reflected in this output. Run 'systmectl daemon-reload' to update to the version shown here."